platform/security/security-oidc-realm/src/main/java/org/codice/ddf/security/oidc/realm/OidcCredentialsResolver.java

```diff
@@ -47,50 +50,62 @@
 import org.pac4j.oidc.client.OidcClient;
 import org.pac4j.oidc.config.OidcConfiguration;
 import org.pac4j.oidc.credentials.OidcCredentials;
-import org.pac4j.oidc.credentials.authenticator.OidcAuthenticator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class CustomOidcAuthenticator extends OidcAuthenticator {
+public class OidcCredentialsResolver {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(CustomOidcAuthenticator.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(OidcCredentialsResolver.class);
 
+  private OidcConfiguration oidcConfiguration;
+  private OidcClient oidcClient;
   private OidcTokenValidator oidcTokenValidator;
   private OIDCProviderMetadata metadata;
 
-  public CustomOidcAuthenticator(
-      OidcConfiguration configuration, OidcClient client, OIDCProviderMetadata metadata) {
-    super(configuration, client);
+  public OidcCredentialsResolver(
+      OidcConfiguration oidcConfiguration, OidcClient oidcClient, OIDCProviderMetadata metadata) {
+    this.oidcConfiguration = oidcConfiguration;
+    this.oidcClient = oidcClient;
     this.metadata = metadata;
-    oidcTokenValidator = new OidcTokenValidator(configuration, metadata);
+    oidcTokenValidator = new OidcTokenValidator(oidcConfiguration, metadata);
   }
 
   /* This methods job is to try and get an id token from a
   1. refresh token
   2. authorization code
   3. access token
   */
-  @Override
-  public void validate(OidcCredentials credentials, WebContext webContext) {
-    final RefreshToken refreshToken = credentials.getRefreshToken();
-    final AuthorizationCode authorizationCode = credentials.getCode();
+  public void resolveIdToken(OidcCredentials credentials, WebContext webContext) {
+    final AccessToken initialAccessToken = credentials.getAccessToken();
+    final JWT initialIdToken = credentials.getIdToken();
+    if (initialIdToken != null) {
+      if (initialAccessToken != null) {
```

Comment:
✏️ We don't need this check here the `validateAccessToken ` will just return if the access token is null
Can just be 
```
oidcTokenValidator.validateAccessToken(initialAccessToken, initialIdToken);
if (initialIdToken != null) {
    oidcTokenValidator.validateIdTokens(initialIdToken, webContext);
}
```
We also want to validate an access token even if the id token is null